x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/ml_inference_logic.ts

```diff
@@ -271,21 +267,25 @@ export const MLInferenceLogic = kea<
       ],
     ],
     values: [
+      CachedFetchModelsApiLogic,
+      ['modelsData', 'status as modelsStatus', 'isInitialLoading as isModelsInitialLoading'],
       CachedFetchIndexApiLogic,
       ['indexData as index'],
       FetchMlInferencePipelinesApiLogic,
       ['data as mlInferencePipelinesData'],
       MappingsApiLogic,
       ['data as mappingData', 'status as mappingStatus'],
-      TrainedModelsApiLogic,
-      ['data as mlModelsData', 'status as mlModelsStatus'],
       FetchMlInferencePipelineProcessorsApiLogic,
       ['data as mlInferencePipelineProcessors'],
       FetchPipelineApiLogic,
       ['data as existingPipeline'],
     ],
   },
-  events: {},
+  events: ({ actions }) => ({
+    afterMount: () => {
+      actions.startPollingModels();
```

Comment:
For some reason the `GET /ml/models` API is called twice upon opening the pipeline config flyout, and populates the models, so this hook is technically not needed. But I can't find what triggers the other call, so I'm leaving it here while I investigate.

Reply:
The 2nd call is caused by `useEffect() -> setIndexName` in `add_inference_pipeline_flyout.tsx`. I removed the listener, moved the fetch operations there, and removed the `afterMount` hook.